Sources/TSCBasic/Path.swift

```diff
@@ -613,14 +612,14 @@ private struct UNIXPath: Path {
     }
 
     init(normalizingRelativePath path: String) {
-      #if os(Windows)
-        var buffer: [WCHAR] = Array<WCHAR>(repeating: 0, count: Int(MAX_PATH + 1))
-        _ = path.replacingOccurrences(of: "/", with: "\\").withCString(encodedAs: UTF16.self) {
-            PathCanonicalizeW(&buffer, $0)
-        }
-        self.init(string: String(decodingCString: buffer, as: UTF16.self))
-      #else
-        precondition(path.first != "/")
+        let pathSeparator: Character
+#if os(Windows)
+        pathSeparator = "\\"
+        let path = path.replacingOccurrences(of: "/", with: "\\")
+#else
+        pathSeparator = "/"
+#endif
+        precondition(path.first != pathSeparator)
```

Comment:
This precondition doesn't hold.  `\path` is valid as a relative path.  It is relative to the current drive.

Reply:
Although it is literally "relative", Windows Shell (and Swift which depends on it) regards `\path` `\?\path` `\\share\path` as absolute paths (according to the response of `PathIsRelativeW`). Personally I think these paths should be treated the same with a regular absolute path. For example, `\path\..\..` indicates `\` like `C:\path\..\..` indicates `C:\`, but `path\..\..` indicates `..`.

Reply:
UNC paths are always absolute.  It is just a drive relative path which is not absolute.  That distinction is important as it impacts the lexical traversal of the path string.

Reply:
`PathIsRelativeW` regards `\path` as an absolute path. Also, if we treat it as relative, behavior regarding `..` will be unaligned.

Reply:
Ping @compnerd

Reply:
I still think that the differences are material and need to be addressed.  Paths in Windows do not behave like Unix, and so you will have differences.  e.g.

path1 = "C:\Users\compnerd"
path2 = "D:"
path1 + path2 is not `C:\Users\compnerd\D:" but rather "C:\Windows\System32\WindowsMetadata\"

because the current directory on D: happened to be "..\..\Windows\System32\WindowsMetadata"

Reply:
I’ll look into more critical cases here. 

However, if I regard `\path` as a relative path, it will change the behavior of a bunch of APIs (while this PR focuses on bug fixes), so that change definitely needs to be addressed in a separate PR.

Reply:
IMO I believe this PR is fully capable of fixing faulty behavior without changing the current judgement of relative paths and absolute paths. To fully support Windows path system, we’d better work on another one which may contain a complete refactoring.

What this PR mainly addressed and fixed is the problem of `..` being ignored, and of crashing on an empty path, which are critical bugs for tools like SwiftPM.